kotlinpoet-metadata-specs/src/main/kotlin/com/squareup/kotlinpoet/metadata/specs/internal/KmTypes.kt

```diff
@@ -173,8 +173,49 @@ internal fun ImmutableKmTypeParameter.toTypeVariableName(
 
 @KotlinPoetMetadataPreview
 private fun ImmutableKmFlexibleTypeUpperBound.toTypeName(
-  typeParamResolver: ((index: Int) -> TypeName)
+  typeParamResolver: TypeParameterResolver
 ): TypeName {
   // TODO tag typeFlexibilityId somehow?
   return WildcardTypeName.producerOf(type.toTypeName(typeParamResolver))
 }
+internal interface TypeParameterResolver {
+  val parametersMap: Map<Int, TypeVariableName>
+  operator fun get(index: Int): TypeVariableName
```

Comment:
Add default implementation?

Reply:
I don't think it's worthwhile, we use this for default argument params so we want a singleton object we can point to